### PR TITLE
Feat improve type annotation in jina 743

### PR DESCRIPTION
**Pull request by rameshwara**
Files updated with changes requested by JoanFM

### CODE REVIEW

File: jina/__init__.py [hunk 1]
```diff
@@ -83,7 +84,7 @@
 
 
 def import_classes(namespace: str, targets=None,
-                   show_import_table: bool = False, import_once: bool = False):
+                   show_import_table: bool = False, import_once: bool = False) -> Union[None, Dict]:
```
Comment (JoanFM) [inline jina/__init__.py:87]: `Union[type, None]` is equivalent to `Optional[type]`

File: jina/docker/database.py [hunk 2]
```diff
@@ -49,13 +49,18 @@ def database(self):
     def collection(self):
         return self.database[self.collection_name]
     
-    def find(self, query: dict):
+    def insert(self, document: str) -> str:
```
Comment (JoanFM) [inline jina/docker/database.py:52]: still a function that should not have been added

File: jina/drivers/__init__.py [hunk 3]
```diff
@@ -265,11 +265,11 @@ def _apply_all(self, docs: Iterable['jina_pb2.Document'], context_doc: 'jina_pb2
     def __call__(self, *args, **kwargs):
         self._traverse_apply(self.req.docs, *args, **kwargs)
 
-    def _traverse_apply(self, docs, *args, **kwargs):
+    def _traverse_apply(self, docs: Iterable['jina_pb2.Document'], *args, **kwargs) -> None:
         """often useful when you delete a recursive structure """
 
-        def post_traverse(_docs, recur_on, context_doc=None,
-                          depth_name='granularity', recur_start=0, recur_end=1):
+        def post_traverse(_docs: List, recur_on: str, context_doc: str=None,
```
Comment (JoanFM) [inline jina/drivers/__init__.py:271]: same here, `_docs` is `Iterable['jina_pb2.Document']`

File: jina/drivers/__init__.py [hunk 4]
```diff
@@ -292,8 +292,8 @@ def post_traverse(_docs, recur_on, context_doc=None,
                 if self._is_apply_all and getattr(_docs[0], depth_name) >= recur_start:
                     self._apply_all(_docs, context_doc, recur_on, *args, **kwargs)
 
-        def pre_traverse(_docs, recur_on, context_doc=None,
-                         depth_name='granularity', recur_start=0, recur_end=1):
+        def pre_traverse(_docs: List, recur_on: str, context_doc: str=None,
```
Comment (JoanFM) [inline jina/drivers/__init__.py:295]: same here

File: jina/executors/indexers/vector.py [hunk 5]
```diff
@@ -2,33 +2,17 @@
...
+from typing import Optional, List, Union, Tuple
 
 import numpy as np
 
 from . import BaseVectorIndexer
-from ..decorators import batching
 from ...helper import cached_property
 
 
 class BaseNumpyIndexer(BaseVectorIndexer):
```
Comment (JoanFM) [inline jina/executors/indexers/vector.py:14]: this file has been changed in a recent merge from master and the update has not been properly done!

File: jina/executors/rankers/__init__.py [hunk 6]
```diff
@@ -83,7 +83,7 @@ def _get_score(self, match_idx, query_chunk_meta, match_chunk_meta, *args, **kwa
         raise NotImplementedError
 
     @staticmethod
-    def sort_doc_by_score(r):
+    def sort_doc_by_score(r: list) -> 'np.ndarray':
```
Comment (JoanFM) [inline jina/executors/rankers/__init__.py:86]: this is `List[Tuple[int, float]]`

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by JoanFM** [comment]
Hello @rameshwara,

I think this is a great contribution, however it is too big for us to handle it properly in one PR, and many changes have not been merged properly in ur branch. Could you please break it down in different PRs, tackling different parts of the code one step at a time?

Plus, check https://github.com/jina-ai/jina/issues/553 and our Contribution guidelines to see how to make the commit messages pass our CI checks.

**Comment by rameshwara** [comment]
> Hello @rameshwara,
> 
> I think this is a great contribution, however it is too big for us to handle it properly in one PR, and many changes have not been merged properly in ur branch. Could you please break it down in different PRs, tackling different parts of the code one step at a time?
> 
> Plus, check #553 and our Contribution guidelines to see how to make the commit messages pass our CI checks.

Hello Joan, I understand it has become too big, would it be fine if I could restart this with smaller pull requests?

**Comment by JoanFM** [comment]
> > Hello @rameshwara,
> > I think this is a great contribution, however it is too big for us to handle it properly in one PR, and many changes have not been merged properly in ur branch. Could you please break it down in different PRs, tackling different parts of the code one step at a time?
> > Plus, check #553 and our Contribution guidelines to see how to make the commit messages pass our CI checks.
> 
> Hello @JoanFM, I understand it has become too big, would it be fine if I could restart this with smaller pull requests?

Sure, that was my idea, closing this PR for now, thanks!

**Pull request closed without merging**